### DISCUSSION TITLE
Help with PHP API POST request

### DISCUSSION THREAD

**Discussion by lukeyD92** [community/community#31923] [Apps, API and Webhooks]
I'm fairly new to coding and I am struggling to get an API POST request to work - it might be a simple fix for someone but I've tried everything and I just cannot figure it out :(

`	public function broadband_plan(){
		$common = array();
		$common['main_menu'] = 'Broadband plan';

		$cli_or_postcode = isset($_GET["cli_or_postcode"]) ? $_GET["cli_or_postcode"] : "";
		$district_id = isset($_GET["district_id"]) ? $_GET["district_id"] : "";
		$sub_premises = isset($_GET["sub_premises"]) ? $_GET["sub_premises"] : "";
		$premises_name = isset($_GET["premises_name"]) ? $_GET["premises_name"] : "";
		$thoroughfare_number = isset($_GET["thoroughfare_number"]) ? $_GET["thoroughfare_number"] : "";
		$thoroughfare_name = isset($_GET["thoroughfare_name"]) ? $_GET["thoroughfare_name"] : "";
		$locality = isset($_GET["locality"]) ? $_GET["locality"] : "";
		$postcode = isset($_GET["postcode"]) ? $_GET["postcode"] : "";
		$nad_key = isset($_GET["nad_key"]) ? $_GET["nad_key"] : "";
		$post_town = isset($_GET["post_town"]) ? $_GET["post_town"] : "";
		$county = isset($_GET["county"]) ? $_GET["county"] : "";
		$district_id = isset($_GET["district_id"]) ? $_GET["district_id"] : "";

		$request = array();
        $request['apiUsername'] = '';
        $request['apiPassword'] = '';
		$request['encryption']   = 'SHA-512';
		$request['platform']     = 'LIVE';
		$request['method']       = 'POST';
		
		$address = '{
			"sub_premises": $sub_premises,
            "premises_name": $premises_name,
            "thoroughfare_number": $thoroughfare_number,
            "thoroughfare_name": $thoroughfare_name,
            "post_town": $post_town,
            "postcode": $postcode,
            "district_id": $district_id,
            "nad_key": $nad_key
		}';
		$address = json_encode($address);

		$request['url'] = "/broadband/availability" . $address;
		
		
	
		$broadband_plan_detail = array();
		if (!empty($address)) {
			$broadband_plan_detail = $this->get_plan($request);
		}

		return view('front.broadband_plan',compact('common','broadband_plan_detail'));
	}`

**Comment by MThos** [comment]
what is the error you are getting?
